salt/_modules/docker_registry.py

```diff
@@ -0,0 +1,87 @@
+'''
+Functions to interact with a Docker registry.
+'''
+
+import logging
+
+import salt.utils.json
+import salt.utils.path
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'docker_registry'
+
+
+def __virtual__():
+    if not salt.utils.path.which('skopeo'):
+        return (False, "'skopeo' command not found")
+
+    return __virtualname__
+
+
+def get_image_manifest(image, tls_verify=True):
+    '''
+    Retrieve an image manifest from a repository.
+
+    .. note::
+
+       This uses the :command:`skopeo` command.
+
+    image
+        Name of the image (e.g. `localhost:5000/metalk8s/nginx:1.14.2`)
+    tls_verify : True
+        Verify TLS certificate of the repository. Set to `False` if the
+        repository uses HTTP instead of HTTPS.
```

Comment:
An overall comment. In salt, since you can extract the docstring with 
```
salt-call  -d docker_registry
```
it's a good practice to put one or more examples in the docstring of how you use your module
```
.. code:: bash

  salt '*' docker_registry.get_image_manifest localhost:5000/metalk8s/nginx:1.14.2
```

Reply:
I know, hence the existing docstrings ;)

Note however these modules are not meant (right now) for any 'public consumption' outside of this project, so I don't want to go overboard with examples and code documentation either.